src/lib/APS2.cpp

```diff
@@ -1,20 +1,27 @@
 #include "APS2.h"
+#include "APS2Datagram.h"
 
-APS2::APS2() :  isOpen{false}, channels_(2), samplingRate_{0} {};
+APS2::APS2() :  isOpen{false}, legacy_firmware{false}, channels_(2), samplingRate_{0} {};
 
-APS2::APS2(string deviceSerial) :  isOpen{false}, deviceSerial_{deviceSerial}, samplingRate_{0} {
+APS2::APS2(string deviceSerial) :  isOpen{false}, legacy_firmware{false}, deviceSerial_{deviceSerial}, samplingRate_{0} {
 	channels_.reserve(2);
 	for(size_t ct=0; ct<2; ct++) channels_.push_back(Channel(ct));
 };
 
 APS2::~APS2() = default;
 
-void APS2::connect(shared_ptr<APSEthernet> && ethernetRM) {
+void APS2::connect(shared_ptr<APS2Ethernet> && ethernetRM) {
+	FILE_LOG(logDEBUG) << "APS2::connect";
+	//Hold on to APS2Ethernet class to keep socket alive
 	ethernetRM_ = ethernetRM;
 	if (!isOpen) {
 		ethernetRM_->connect(deviceSerial_);
+		//Figure out whether this is an APS2 or TDM and what register map to use
 		try {
 			APSStatusBank_t statusRegs = read_status_registers();
+			if ( statusRegs.userFirmwareVersion != 0xbadda555 ) {
```

Comment:
But I have sometimes shoved random stuff into that register for debugging purposes. Can we just check the major firmware version number (assuming you rolled to 3.x...)?